### PR TITLE
EXT-151 Initial build/commit.

### PR DESCRIPTION
**Pull request by Camwyn**
Note the original is the extension template.
📷 http://p.tri.be/JynyYi
📸 http://p.tri.be/E6a7rm

**REQUIRES** https://github.com/moderntribe/the-events-calendar/pull/3135

[EXT-151]

[EXT-151]: https://moderntribe.atlassian.net/browse/EXT-151

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -1,21 +1,20 @@
-=== [Base Plugin Name] Extension: [Extension Name] ===
+=== [Base Plugin Name] Extension: Open external links in a new tab ===
```
Comment (juanfra) [inline readme.txt]:
```suggestion
=== The Events Calendar Extension: Open external links in a new tab ===
```

File: src/Settings.php [hunk 2]
```diff
@@ -113,6 +110,25 @@ public function get_option( $key = '', $default = '' ) {
...
 
+		/**
+		 * Given a link-type, get this extension's option value for it.
+		 *
+		 * This automatically prepends this extension's option prefix so you can just do `$this->get_option( 'a_setting' )`.
+		 *
+		 * @see tribe_get_option()
+		 *
+		 * @param string $key
+		 *
```
Comment (juanfra) [inline src/Settings.php:135]: Maybe add `@since`?

File: src/Settings.php [hunk 3]
```diff
@@ -113,6 +110,25 @@ public function get_option( $key = '', $default = '' ) {
...
 		}
 
+		/**
+		 * Given a link-type, get this extension's option value for it.
+		 *
+		 * This automatically prepends this extension's option prefix so you can just do `$this->get_option( 'a_setting' )`.
+		 *
+		 * @see tribe_get_option()
+		 *
+		 * @param string $key
```
Comment (juanfra) [inline src/Settings.php]: Wrong var in docblock.

File: src/Settings.php [hunk 4]
```diff
@@ -256,21 +254,19 @@ function ( $key ) {
 		}
 
 		/**
-		 * Here is an example of getting some HTML for the Settings Header.
-		 *
-		 * TODO: Delete this method if you do not need a heading for your settings. Also remove the corresponding element in the the $fields array in the `add_settings()` method above.
+		 * Build options to present to user
 		 *
```
Comment (juanfra) [inline src/Settings.php:286]: Maybe add `@since`?

**Review by juanfra — APPROVED** [review]
🚀 Looking great! I'd set the date of the changelog before merging.

File: readme.txt [hunk 5]
```diff
@@ -37,7 +36,12 @@ We're always interested in your feedback and our [Help Desk](https://support.the
 
 == Changelog ==
 
+= [1.0.2] TBD =
+
+* Actual working code.
+
 = [1.0.1] 2020-04-09 =
+
 * Test version change.
```
Comment (cliffordp) [inline readme.txt]: eww ;)

Can we delete the remote tag? Run this on the repo in your Terminal and then re-release actual 1.0.1 in the changelog with accurate release date.

`git push --delete origin 1.0.1`

File: tribe-ext-external-links-in-new-tab.php [hunk 6]
```diff
@@ -41,54 +41,38 @@ class Main extends Tribe__Extension {
...
-			 * All these version numbers are the ones on or after November 16, 2016, but you could remove the version
-			 * number, as it's an optional parameter. Know that your extension code will not run at all (we won't even
-			 * get this far) if you are not running The Events Calendar 4.3.3+ or Event Tickets 4.3.3+, as that is where
-			 * the Tribe__Extension class exists, which is what we are extending.
-			 *
-			 * If using `tribe()`, such as with `Tribe__Dependency`, require TEC/ET version 4.4+ (January 9, 2017).
-			 */
-			// $this->add_required_plugin( 'Tribe__Tickets__Main', '4.4' );
-			// $this->add_required_plugin( 'Tribe__Tickets_Plus__Main', '4.3.3' );
-			// $this->add_required_plugin( 'Tribe__Events__Main', '4.4' );
```
Comment (cliffordp) [inline tribe-ext-external-links-in-new-tab.php:78]: So this plugin doesn't require TEC?

File: tribe-ext-external-links-in-new-tab.php [hunk 7]
```diff
@@ -147,15 +135,28 @@ public function init() {
...
+			if ( ! empty( $this->get_type_option( 'website' ) ) ) {
+				add_filter( 'tribe_get_event_website_link_target', [ $this, 'return_blank_link_target' ] );
+			}
+
+			if ( ! empty( $this->get_type_option( 'organizer' ) ) ) {
+				add_filter( 'tribe_get_event_organizer_link_target', [ $this, 'return_blank_link_target' ] );
+			}
+
+			if ( ! empty( $this->get_type_option( 'content' ) ) ) {
+				add_filter('the_content', [ $this, 'open_content_links_in_new_tab' ], 999);
```
Comment (cliffordp) [inline tribe-ext-external-links-in-new-tab.php]: spaces... run code reformat on the file to be sure/consistent

File: readme.txt (hunk 5 above)
Comment (cliffordp) [reply readme.txt]: Actually, all the `since` tags are v1.0.0 so just make sure github releases/tags are accurate along with changelog

**Review by cliffordp — CHANGES_REQUESTED** [review]

File: readme.txt (hunk 5 above)
Comment (Camwyn) [reply readme.txt]: Yep - was using it to test the GitHub updater plugin for Courtney's article. (she needed screenshots)

Comment (Camwyn) [reply readme.txt]: I deleted it. I will create a new release once we go live with this.

File: tribe-ext-external-links-in-new-tab.php [hunk 8]
```diff
@@ -147,15 +136,28 @@ public function init() {
...
+			if ( ! empty( $this->get_type_option( 'website' ) ) ) {
+				add_filter( 'tribe_get_event_website_link_target', [ $this, 'return_blank_link_target' ] );
+			}
+
+			if ( ! empty( $this->get_type_option( 'organizer' ) ) ) {
+				add_filter( 'tribe_get_event_organizer_link_target', [ $this, 'return_blank_link_target' ] );
+			}
+
+			if ( ! empty( $this->get_type_option( 'content' ) ) ) {
+				add_filter( 'the_content', [ $this, 'open_content_links_in_new_tab' ], 999);
```
Comment (cliffordp) [inline tribe-ext-external-links-in-new-tab.php]:
```suggestion
				add_filter( 'the_content', [ $this, 'open_content_links_in_new_tab' ], 999 );
```

**Review by cliffordp — CHANGES_REQUESTED** [review]

**Review by cliffordp — APPROVED** [review]

**Pull request merged**